quickwit/quickwit-search/src/lib.rs

```diff
@@ -139,34 +138,23 @@ async fn list_relevant_splits(
         .collect::<Vec<_>>())
 }
 
-/// Converts a `LeafHit` into a `Hit`.
-///
-/// Splits may have been created with different DocMappers.
-/// For this reason, leaves are returning a document that is -as much
-/// as possible-, `DocMapper` agnostic.
+/// Converts a leaf document into a JSON String before sending
```

Comment:
what is a leaf document? I would use `NamedFieldDocument` and optionally add the fact that it is coming from tantivy, tantivy `NamedFieldDocument` for example.

Reply:
It's important to mention that the conversion maps the tantivy `NamedFieldDocument` to the targeted schema defined by the `DocMapper`.

I did not really understand the comments. Here is what I would write:

`This conversion is performed at the leaf level only, it avoids having another intermeidate json format between the root and the leaves.`